ddtrace/tracer/tracer.go

```diff
@@ -306,16 +316,50 @@ func (t *tracer) worker(tick <-chan time.Time) {
 	}
 }
 
-func (t *tracer) pushTrace(trace []*span) {
+// traceInfo contains span list and sampling decision of the trace
+// needed to run single span sampling.
+type traceInfo struct {
+	spans            []*span
+	samplingDecision samplingDecision
```

Comment:
```suggestion
	decision samplingDecision
```
Perhaps this is enough?